src/Service.Tests/Snapshots/ConfigurationTests.TestReadingRuntimeConfigForMsSql.verified.txt

```diff
@@ -0,0 +1,2857 @@
+﻿{
+  DataSource: {
+    DatabaseType: MSSQL,
+    Options: {
+      set-session-context: {
+        ValueKind: True
+      }
+    }
+  },
+  Runtime: {
+    Rest: {
+      Enabled: true,
```

Comment:
For my understanding, 

When a config file without the `Methods` section is deserialized, the corresponding `Entity` object has all the 5 HTTP methods populated. So, it appears that we would need the `Methods` section here. 

But, the tests seem to be passing without them. Could you please clarify.

Reply:
@severussundar I am not able to reproduce the behavior you're seeing in this PR. Can you pull latest, Clean/Re-build solution, and then validate you experience the behavior you documented?

I re-ran:
`dab add getbooks --source "getbooks" --source.type "stored-procedure" --permissions "anonymous:execute" --graphql.operation "query"`
then
`dab update getbooks --rest.methods "get"`

and the graphql section was preserved with`"operation":"mutation"`

Reply:
Hey @seantleonard, this doubt/comment is not related to the earlier issue related to updating rest methods.

Thanks for reproing the issue, I'll tried it with the Nuget that gets generated in the build pipeline. I'll try this out once more.

Reply:
gotcha, the longer comment of the earlier issue was not a thread, so couldn't continue there.

Reply:
I repro'd with the build artifact nuget package.

Reply:
Back to the original comment of this thread, I circled back to the Execute for Stored Procedures RFC, and REST methods should default to POST if not defined, and GraphQL default to Mutation if not defined. 
https://github.com/Azure/data-api-builder/issues/1095

Reply:
Should `methods` exist at the runtime level? I don't think it previously did

Reply:
No, the `methods` needn't exist at runtime level for tables and views as we only have the option of either enabling or disabling REST endpoints. Only in case of stored procedures, the field is relevant as it could be used to express the HTTP methods for which the stored procedure is enabled. When stored procedures entities are deserialized, `methods` do get populated. And this snapshot file contains the appropriate `methods` field for sp entities.

I am working on a bug related to rest methods for stored procedures which requires that the `methods` be present at runtime - to distinguish between a) default methods for tables/views vs stored procedures
and b) custom methods entered explicitly via config vs default methods to be used for stored procedures.
When experimenting something for that, it led me to think that we would need the `methods` to be populated here as well... 

The current snapshot file that is present is accurate. Should be good to close this thread.